x/qgb/keeper/keeper_data_commitment.go

```diff
@@ -13,10 +13,16 @@ import (
 // GetCurrentDataCommitment creates the latest data commitment at current height according to
 // the data commitment window specified
 func (k Keeper) GetCurrentDataCommitment(ctx sdk.Context) (types.DataCommitment, error) {
-	beginBlock := uint64(ctx.BlockHeight()) - k.GetDataCommitmentWindowParam(ctx) + 1
-	// for a data commitment window of 400, the ranges will be: 1-400;401-800;801-1200
-	endBlock := uint64(ctx.BlockHeight())
-
+	var beginBlock, endBlock uint64
+	if ctx.BlockHeader().Version.App == 0 {
+		beginBlock = uint64(ctx.BlockHeight()) - k.GetDataCommitmentWindowParam(ctx)
+		// for a data commitment window of 400, the ranges will be: 0-399;400-799;800-1199
+		endBlock = uint64(ctx.BlockHeight()) - 1
+	} else {
+		beginBlock = uint64(ctx.BlockHeight()) - k.GetDataCommitmentWindowParam(ctx) + 1
```

Comment:
[not blocking] is there a risk of underflow here because this logic performs subtraction with `uint64`?

[unrelated] it would be helpful if https://github.com/SweeXordious/celestia-app/blob/aaa09a5853da9956320775404707b9115e02ee54/x/qgb/types/genesis.pb.go#L28 included a doc comment. Something like

```go
	// DataCommitmentWindow is the number of blocks...
	DataCommitmentWindow uint64 `protobuf:"varint,1,opt,name=data_commitment_window,json=dataCommitmentWindow,proto3" json:"data_commitment_window,omitempty"`
```

Reply:
If called correctly, there shouldn't be any underflow. But yes, I guess  at some point we should switch to using int64, and taking negative values as bad values